parsers/ELEXON.py

```diff
@@ -224,11 +271,30 @@ def parse_production_FUELINST(
         lambda x: datetime_from_date_sp(x["Settlement Date"], x["Settlement Period"]),
         axis=1,
     )
-    df = df.groupby(df.columns, axis=1).sum()
+    return df.set_index("datetime")
+
+
+def parse_additional_eso_production(raw_data: List[dict]) -> pd.DataFrame:
+    """Parse additional eso data for embedded wind/solar and hydro storage."""
+    df = pd.DataFrame.from_records(raw_data)
+    df["datetime"] = df.apply(
+        lambda x: datetime_from_date_sp(x["SETTLEMENT_DATE"], x["SETTLEMENT_PERIOD"]),
+        axis=1,
+    )
+    df.rename(columns=ESO_FUEL_MAPPING, inplace=True)
+    return df.set_index("datetime")
+
 
+def process_production_events(
+    fuel_inst_data: pd.DataFrame, eso_data: pd.DataFrame
+) -> List[dict]:
+    """Combine FUELINST report and ESO data together to get the full picture and to EM Format."""
+    df = fuel_inst_data.join(eso_data, rsuffix="_eso")
+    df.rename(columns={"wind_eso": "wind", "solar_eso": "solar"}, inplace=True)
```

Comment:
nitpick, but we generally avoid doing inplace operations

Reply:
So that means we discard completely wind and solar from bnreports in favour of this new source?

Reply:
Elements for why inplace is not really recommended in most cases:

https://towardsdatascience.com/why-you-should-probably-never-use-pandas-inplace-true-9f9f211849e4


TL;DR is that it creates mutability issues

Reply:
@pierresegonne we need to investigate which source we keep, once BNReports comes back online. The source might actually not report embedded wind and solar as well.